packages/@react-spectrum/dropzone/src/DropZone.tsx

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaLabelingProps, DOMProps, DOMRef, StyleProps} from '@react-types/shared';
+import {classNames, SlotProvider, useDOMRef, useStyleProps} from '@react-spectrum/utils';
+import {DropZoneProps, DropZone as RACDropZone} from 'react-aria-components';
+import {mergeProps, useId} from '@react-aria/utils';
+import React, {ReactNode} from 'react';
+import styles from '@adobe/spectrum-css-temp/components/dropzone/vars.css';
+
+export interface SpectrumDropZoneProps extends DropZoneProps, DOMProps, StyleProps, AriaLabelingProps {
+  /** The content to display in the drop zone. */
+  children: ReactNode,
+  /** Whether the dropzone has been filled. */
+  isFilled?: boolean, 
+  /** The message to replace the default banner message that is shown when the drop zone is filled. */
+  replaceMessage?: string
+}
+
+function DropZone(props: SpectrumDropZoneProps, ref: DOMRef<HTMLDivElement>) {
+  let {children, isFilled, replaceMessage, ...otherProps} = props;
+  let {styleProps} = useStyleProps(props);
+  let domRef = useDOMRef(ref);
+  let messageId = useId();
+
+  return (
+    <RACDropZone
+      {...mergeProps(otherProps)}
+      {...styleProps as Omit<React.HTMLAttributes<HTMLElement>, 'onDrop'>}
+      aria-labelledby={isFilled ? messageId : null}
+      className={
+      classNames(
+        styles,
+        'spectrum-Dropzone',
+        styleProps.className
+      )} 
+      ref={domRef}>
+      {({isDropTarget}) => (
```

Comment:
if we do banner display: none in CSS and then only set it to display block when it's drop target, then we don't need this render prop
we could also add a top level state `.spectrum-Dropzone--filled` so we could get rid of the conditional rendering below completely